essay/swar/latency.md

```diff
@@ -149,32 +173,45 @@ multiply128_to128_bits(unsigned __int128, unsigned __int128):           # @multi
         ret
 ```
 
-[^2] We're beginning to work on explaining the basics of processor instructions and operations costs.
+[^2]: We're beginning to work on explaining the basics of processor instructions and operations costs.
+
+### Round trips between SIMD and GPR
+
+[^3]: See: [SIMD-GPR round trips](#round-trips-between-simd-and-gpr)
+
+The SIMD architectures I know all turn lane-wise comparisons such as "greater than" into SIMD registers with each lane full of 1s or 0s depending on the comparison.  This can be used for further SIMD computation.  The problem arises when you want to apply a "horizontal" operation on the SIMD, for example, "what is the first, or least significant lane in which the condition is true?", as in `strlen`, "what is the index of the lane with the first null byte?"  Typically, the SIMD ISA does **not** have an instruction for this, forcing you to create a "mask" extracting bits from each lane, as in x86_64 "vpmovmskb", then process the mask using GPRs, scalar, registers.  If you then want to convert your scalar result back to a SIMD value/register, for further branchless SIMD processing, you are very much like going up a creek without a paddle.
```

Comment:
s/you are very much like going up a creek/you are going up a creek/
or possibly /you are absolutely going up a creek/